pkg/scheduler/framework/types.go

```diff
@@ -40,6 +40,61 @@ import (
 
 var generation int64
 
+// ActionType is an integer to represent one type of resource change.
+// Different ActionTypes can be bit-wised to compose new semantics.
+type ActionType int64
+
+// Constants for ActionTypes.
+const (
+	Add    ActionType = 1 << iota // 1
+	Delete                        // 10
+	// UpdateNodeXYZ is only applicable for Node events.
+	UpdateNodeUnschedulable // 100
+	UpdateNodeAllocatable   // 1000
+	UpdateNodeLabel         // 10000
+	UpdateNodeTaint         // 100000
+	UpdateNodeCondition     // 1000000
+
+	All ActionType = 1<<iota - 1 // 1111111
+
+	// Use the general Update type if you don't either know or care the specific sub-Update type to use.
+	Update = UpdateNodeUnschedulable | UpdateNodeAllocatable |
+		UpdateNodeLabel | UpdateNodeTaint | UpdateNodeCondition
+)
+
+// GVK is short for group/version/kind, which can uniquely represent a particular API resource.
+type GVK string
+
+// Constants for GVKs.
+const (
+	Pod                   GVK = "Pod"
```

Comment:
can we avoid redefining those constants? surely they are defined somewhere in k8s?

Reply:
Unfortunately no, at least for literals.

BTW: I tried to pass in `&v1.Pod{}` and let its GVK auto-detected (). But that needs to be associated with a complete scheme registration, which may be an overkill and also doesn't work in UT/Integration test. I also [asked](https://kubernetes.slack.com/archives/C0EG7JC6T/p1610753647014900) in API machinery slack, no luck on this direction.